### PR TITLE
feat: add optionbuttons to app

### PR DESCRIPTION
**Pull request by DorMosheZarka**
Introduced the OptionButtons component to the application, offering users interactive sorting and filtering capabilities. This addition aims to enhance user navigation and improve access to content.

### CODE REVIEW

**Review by noammu — CHANGES_REQUESTED** [review]
This PR is almost empty... 🙃 (you also probably forgot to merge from main 😉 )

File: src/components/SortByButton/SortByButton.module.css [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   height: 4em;
+   font-weight: 300;
+   margin-right: 1em;
+}
+
+.button {
+   font-size: 1rem;
+   background-color: white;
+   border: 0.1em solid black;
+   border-radius: 0.7em;
```
Comment (noammu) [inline src/components/SortByButton/SortByButton.module.css]: In borders we use pixels.
Just a reminder to see the video about px and em

File: src/components/SortByButton/SortByButton.module.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+.button.open {
+   border-bottom: none;
+   border-bottom-left-radius: 0;
+   border-bottom-right-radius: 0;
+}
+
+.optionsList {
+   list-style: none;
+   background-color: white;
```
Comment (noammu) [inline src/components/SortByButton/SortByButton.module.css]: Are this 2 needed?

File: src/components/SortByButton/SortByButton.module.css [hunk 3]
```diff
@@ -0,0 +1,47 @@
+.sortByButton {
+   width: 25%;
+   height: 4em;
+   font-weight: 300;
+   margin-right: 1em;
+}
```
Comment (noammu) [inline src/components/SortByButton/SortByButton.module.css]: In the future, are you going to give each kind of option-button margin-right just so you have a... ```gap``` between them?

File: src/components/OptionButtons/OptionButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+function OptionButtons() {
+   return (
+      <div className={styles.optionButtons}>
+         <SortByButton />
+      </div>
+   );
+}
+
+export default OptionButtons;
```
Comment (noammu) [inline src/components/OptionButtons/OptionButtons.tsx]: In general, this approach is ok.
But what will happen if in the future you will add more option buttons (that you currently don't think about them)?
Will need to go over each one of the buttons and change their style just so you could add the extra button well?
What about if this component was mainly just a wrapper which defines the spacing of this area in relation to other areas, and also the spacing to have between the buttons (like with a gap)?

Comment (noammu) [inline src/components/OptionButtons/OptionButtons.tsx]: Another idea is, what if someone would want to have this component, but without one of the buttons that this component will have? 
How can you solve this problem? (generalize the component, children props 😉 )

File: src/components/SortByButton/SortByButton.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                     {option}
+                  </li>
+               ))}
+            </ul>
+         )}
+      </div>
+   );
+}
+
+export default SortByButton;
```
Comment (noammu) [inline src/components/SortByButton/SortByButton.tsx]: NICE!

**Review by noammu — CHANGES_REQUESTED** [review]

File: src/components/SortByButton/SortByButton.module.css (hunk 2 above)
Comment (DorMosheZarka) [reply src/components/SortByButton/SortByButton.module.css]: Its hard to see but the default is transparent. Apparently, display: flex of each list item cancel the list bullets...Thanks!

File: src/components/OptionButtons/OptionButtons.tsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import styles from './OptionButtons.module.css';
+
+function OptionButtons({ children }: { children: React.ReactNode }) {
+   return <div className={styles.optionButtons}>{children}</div>;
+}
+
+export default OptionButtons;
```
Comment (noammu) [inline src/components/OptionButtons/OptionButtons.tsx]: This is ok.
But in the future you would probably have multiple props, and then writing it like this wouldn't look great to say the least.
How about we change it to ```interface```? Would be a good exercise and we have tons of them in Codux :)

File: src/components/SortByButton/SortByButton.module.css [hunk 7]
```diff
@@ -0,0 +1,45 @@
+.sortByButton {
+   width: 25%;
+   height: 4em;
+   font-weight: 300;
+}
```
Comment (noammu) [inline src/components/SortByButton/SortByButton.module.css]: Is the ```font-weight``` needed? my ui eyes are not the best so you tell me 😆

File: src/components/SortByButton/SortByButton.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+   };
+
+   return (
+      <div className={styles.sortByButton}>
+         <button
+            className={`${styles.button} ${isOpen ? styles.open : null}`}
+            onClick={toggleIsOpen}>
+            Sort By: {selectedOption} <span>{isOpen ? '▲' : '▼'}</span>
+         </button>
+         {isOpen && (
```
Comment (noammu) [inline src/components/SortByButton/SortByButton.tsx]: Why use ```<span>```?

**Review by noammu — CHANGES_REQUESTED** [review]

File: src/components/SortByButton/SortByButton.module.css (hunk 7 above)
Comment (DorMosheZarka) [reply src/components/SortByButton/SortByButton.module.css]: indeed, it is necessary, but I've moved it to optionItems for better readability.

**Review by noammu — APPROVED** [review]
Noice

**Pull request merged**